nrf_modem/include/nrf_socket.h

```diff
@@ -17,7 +17,22 @@
 extern "C" {
 #endif
 
+#if !defined(__GNUC__) || (__GNUC__ == 0)
 typedef int32_t ssize_t;
+#else
+#include <sys/types.h>
```

Comment:
I am not sure about adding a dependency on a POSIX header.
What about checking for `_SSIZE_T_DEFINED` instead?

Reply:
Do you think this would work @tothero? Does your `sys/types.h` file define this macro?